### PR TITLE
RFR: Resolve is anonymised failure for pydicom 1.2.0

### PR DESCRIPTION
**Pull request by Centrus007**
Improve DICOM dataset equality test in `test_dicom_anonymise()`.

### CODE REVIEW

File: tests/test_dicom/test_dicom_anonymise.py [hunk 1]
```diff
@@ -33,10 +32,9 @@ def test_is_anonymised_dataset():
     ds_anon_func = anonymise_dataset(ds, delete_private_tags=False,
                                      copy_dataset=True)
 
-    # Patient Name should've been the only non-anonymous value. If not,
-    # test data has been changed
-    for elem_manual, elem_func in zip(ds_anon_manual, ds_anon_func):
-        assert elem_manual.value == elem_func.value
+    elems_manual = sorted(list(ds_anon_manual.iterall()), key = lambda x: x.tag)
```
Comment (Centrus007) [inline tests/test_dicom/test_dicom_anonymise.py]: `ds.iterall()` skips Sequences, which (along with other things) were causing issues in `ds.__eq__()`

**Comment by Centrus007** [comment]
(Just wanted to sneak this in which it was front-of-mind).

File: tests/test_dicom/test_dicom_anonymise.py (hunk 1 above)
Comment (Centrus007) [reply tests/test_dicom/test_dicom_anonymise.py]:
```suggestion
    elems_anon_manual = sorted(list(ds_anon_manual.iterall()), key = lambda x: x.tag)
```

File: tests/test_dicom/test_dicom_anonymise.py [hunk 2]
```diff
@@ -32,12 +33,9 @@ def test_is_anonymised_dataset():
...
                                      copy_dataset=True)
 
-    # Patient Name should've been the only non-anonymous value. If not,
-    # test data has been changed
-    for elem_manual, elem_func in zip(ds_anon_manual, ds_anon_func):
-        # Assumes all non-private, identifying tags are top-level
-        if elem_manual.VR != 'SQ':
-            assert elem_manual == elem_func
+    elems_manual = sorted(list(ds_anon_manual.iterall()), key = lambda x: x.tag)
+    elems_func = sorted(list(ds_anon_func.iterall()), key = lambda x: x.tag)
```
Comment (Centrus007) [inline tests/test_dicom/test_dicom_anonymise.py]:
```suggestion
    elems_anon_func = sorted(list(ds_anon_func.iterall()), key = lambda x: x.tag)
```

File: tests/test_dicom/test_dicom_anonymise.py [hunk 3]
```diff
@@ -32,12 +33,9 @@ def test_is_anonymised_dataset():
...
+    assert elems_manual == elems_func
```
Comment (Centrus007) [inline tests/test_dicom/test_dicom_anonymise.py]:
```suggestion
    assert elems_anon_manual == elems_anon_func
```

**Comment by SimonBiggs** [comment]
All good, git is pretty awesome at dealing with this sort of stuff :)

**Pull request merged**